### PR TITLE
rivalcfg: init at 4.3.0

### PR DESCRIPTION
**Pull request by warmdarksea**
###### Motivation for this change
rivalcfg is a utility program for changing some settings on SteelSeries brand computer mice, including sensor sensitivity, LED lighting, lighting effects, and button mapping.

I just realized I had an old issue https://github.com/NixOS/nixpkgs/pull/66534 where I tried to merge this same package - should I try to revive that one? I think I've fixed all the issues with it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/rivalcfg/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    owner = "flozz";
+    repo = "rivalcfg";
+    rev = "v${version}";
+    sha256 = "sha256:1jms1dc342cxhagbrnjq130zkw0gjqggx33iy44d4pnxrz873wgz";
+  };
+
+  propagatedBuildInputs = [ hidapi setuptools ];
+
+  checkInputs = [ pytest ];
+  checkPhase = "pytest";
```
Comment (SuperSandro2000) [inline pkgs/misc/rivalcfg/default.nix:17]: Please use pytestCheckHook.

File: pkgs/misc/rivalcfg/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  # tests are broken
```
Comment (SuperSandro2000) [inline pkgs/misc/rivalcfg/default.nix:19]: How they are broken?

File: pkgs/misc/rivalcfg/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  doCheck = false;
+
+  # this file has to be copied here instead of generated at build time because
+  # rivalcfg --update-udev will fail if it detects a supported device but cannot
+  # access it
+  # it should probably be regenerated on version bumps
+  postInstall = ''
+    set -x
```
Comment (SuperSandro2000) [inline pkgs/misc/rivalcfg/default.nix:27]:
```suggestion
```
this can be achieve by suppling -v to nix-build.

**Comment by colemickens** [comment]
@ornxka do you want to finish this up? I have an updated version that I can send otherwise...

**Pull request closed without merging**